lib/sdk.js

```diff
@@ -0,0 +1,414 @@
+var bz2 = require('unbzip2-stream');
+var createHash = require('sha.js');
+var fs = require('fs-extra-promise');
+var osenv = require('osenv');
+var path = require('path');
+var Progress = require('progress');
+var Promise = require('bluebird');
+var request = require('request');
+var spawn = require('child_process').spawn;
+var tar = require('tar-fs');
+var tmp = require('tmp');
+var Transform = require('stream').Transform;
+var zlib = require('zlib');
+
+var log = require('./log');
+
+var getPlatform = exports.getPlatform = () => {
+  switch (process.platform) {
+    case 'darwin':
+      return 'macos';
+    case 'linux':
+      return 'linux';
+    default:
+      throw new Error('Your platform is not yet supported for cross-compilation.');
+  }
+};
+
+var paths = {
+  sdk: path.join(osenv.home(), '.tessel/sdk'),
+  rustlib: path.join(osenv.home(), '.tessel/rust'),
+};
+
+var sdkPath = {
+  'macos': 'https://builds.tessel.io/t2/sdk/t2-sdk-macos-x86_64.tar.bz2',
+  'linux': 'https://builds.tessel.io/t2/sdk/t2-sdk-linux-x86_64.tar.bz2',
+};
+
+var rustlibUrl = 'https://builds.tessel.io/t2/sdk/t2-rustlib-VERSION.tar.gz';
+
+function sha256stream() {
+  var sha256 = createHash('sha256');
+  var stream = new Transform();
+  stream._transform = function(chunk, encoding, callback) {
+    this.push(chunk);
+    sha256.update(chunk);
+    callback();
+  };
+  stream.on('finish', function() {
+    stream.emit('sha256', sha256.digest('hex'));
+  });
+  return stream;
+}
+
+function sha256file(hash, name) {
+  return hash + '  ' + name + '\n';
+}
+
+function download(url) {
+  var req = request.get(url);
+
+  // When we receive the response
+  req.on('response', (res) => {
+
+    // Parse out the length of the incoming bundle
+    var contentLength = parseInt(res.headers['content-length'], 10);
+
+    // Create a new progress bar
+    var bar = new Progress('     [:bar] :percent :etas remaining', {
+      clear: true,
+      complete: '=',
+      incomplete: ' ',
+      width: 20,
+      total: contentLength
+    });
+
+    // When we get incoming data, update the progress bar
+    res.on('data', (chunk) => {
+      bar.tick(chunk.length);
+    });
+  });
+
+  return req;
+}
+
+function downloadString(url) {
+  return new Promise((resolve, reject) => {
+    request({
+      url: url,
+      // We want to force Cloudfront to serve us the latest file.
+      headers: {
+        'Accept-Encoding': 'gzip, deflate',
+      },
+    }, (error, response, body) => {
+      if (!error && response.statusCode === 200) {
+        resolve(body);
+      } else {
+        reject(error || response.statusCode);
+      }
+    });
+  });
+}
+
+var tmpdir = exports.tmpdir = () => {
+  return new Promise((resolve, reject) => {
+    tmp.dir(function(err, tmppath, cleanup) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve({
+          path: tmppath,
+          cleanup: cleanup,
+        });
+      }
+    });
+  });
+};
```

Comment:
Why is this exported? (also, why is everything exported on `exports` and not `module.exports`, per the project convention?